AnoopExercises/Sources/SearchVC.swift

```diff
@@ -5,9 +5,189 @@
 import UIKit
 
 class SearchVC: UIViewController {
+    // MARK: - Variables
+
+    private weak var searchBar: UISearchBar!
+    private weak var searchButton: UIButton!
+    var activityIndicator: UIActivityIndicatorView!
+    var activeTextField: UISearchBar?
+    var activeButton: UIButton?
+
+    // MARK: - View Life Cycle
+
     override func viewDidLoad() {
         super.viewDidLoad()
-        view.backgroundColor = R.color.systemBackground()
-        navigationItem.title = "Flickr"
+        view.backgroundColor = UIColor(named: "systemBackground")
+        setupTitle()
+
+        let tap = UITapGestureRecognizer(target: self, action: #selector(dismissKeyboard))
+        view.addGestureRecognizer(tap)
+
+        var constraints = setupSearchBar()
+        constraints += setupSearchButton()
+        constraints += setupActivityIndicator()
+
+        searchButton.addTarget(self, action: #selector(beginSearch), for: .touchUpInside)
+
+        NSLayoutConstraint.activate(constraints)
+
+        NotificationCenter.default.addObserver(
+            self,
+            selector: #selector(keyboardWillShow(notification:)),
+            name: UIResponder.keyboardWillShowNotification, object: nil
+        )
+        NotificationCenter.default.addObserver(
+            self,
+            selector: #selector(keyboardWillHide(notification:)),
+            name: UIResponder.keyboardWillHideNotification,
+            object: nil
+        )
+    }
+
+    // MARK: - Setup
+
+    private func setupSearchBar() -> [NSLayoutConstraint] {
+        let searchBar = UISearchBar()
+        searchBar.configureView { sb in
```

Comment:
```suggestion
        searchBar.configureView { sB in
```
Good to have camelCasing

Reply:
okay ill make the changes